servicesCore/requestCore_service.py

```diff
@@ -1,35 +1,219 @@
+    """
+
     def __init__(self):
-        pass
+        self._web3Single = Web3Single.getInstance()
+        self._ipfs = Ipfs.getInstance()
+        # TODO: load requestCoreArtifact using JSONLoader
+        self._abiRequestCore = requestCoreArtifact['abi']
+        if not requestCoreArtifact['networks'][self._web3Single.networkName]:
+            raise ValueError('RequestCore Artifact does not have configuration for network: "' + self._web3Single.networkName + '"')
+        self._addressRequestCore = requestCoreArtifact['networks'][self._web3Single.networkName].address
+        self._instanceRequestCore = self._web3Single.web3.eth.Contract(self._abiRequestCore, self._addressRequestCore)
 
-    def getCurrentNumRequest(self):
-        pass
+    async def getCurrentNumRequest(self):
+        """
+        Get the number of the last request (N.B. != id)
+        """
+        try:
+            return self._instanceRequestCore.call().numRequests()
+        except Exception as e:
+            raise e
 
-    def getVersion(self):
-        pass
+    async def getVersion(self):
+        """
+        Get the version of the contract
+        """
+        try:
+            return self._instanceRequestCore.call().VERSION()
+        except Exception as e:
+            raise e
 
-    def getCollectEstimation(self, expectedAmount: any, currencyContract: str, extension: str):
-        pass
+    async def getCollectEstimation(self, expectedAmount: any, currencyContract: str, extension: str):
+        """
+        Get the estimation of ether (in wei) needed to create a request
+        :param expectedAmount: amount expected of the request
+        :param currencyContract: address of the currency contract of the request
+        :param extension: address of the extension contract of the request
+        """
+        if not self._web3Single.isAddressNoChecksum(currencyContract):
+            raise ValueError('currencyContract must be a valid eth address')
 
-    def getRequest(self, requestId: str):
-        pass
+        if extension and extension != '' and not self._web3Single.isAddressNoChecksum(extension):
+            raise ValueError('extension must be a valid eth address')
+
+        try:
+            data = self._instanceRequestCore.call().getCollectEstimation(expectedAmount, currencyContract, extension)
+            return data
+        except Exception as e:
+            raise e
+
+    async def getRequest(self, requestId: str):
+        """
+        Get a request by its requestId
+        :param requestId: requestId of the request
+        """
+        if not self._web3Single.isHexStrictBytes32(requestId):
+            raise ValueError('requestId must be a 32 bytes hex string')
+        try:
+            data = self._instanceRequestCore.call().requests(requestId)
+            if data.creator == EMPTY_BYTES_32:
+                raise ValueError('request not found')
+
+            # excluding BN
+            dataResult = {
```

Comment:
Is data here a data structure or a dictionary?

Reply:
Unclear.  The `web3` documentation isn't super explicit (I think v4 changed the Contract api).

Reply:
Then we'll assume that it's fine and if it errors out later it's an easy fix.